src/common/NavBar.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from "react";
+import { Link } from "react-router-dom";
+
+class NavBar extends Component{
+  constructor(props){
+    super(props)
+    this.state = {isAuthenticated: this.props.isAuthenticated}
+    this.Signin = this.Signin.bind(this)
+    this.SignOut = this.SignOut.bind(this)
+  }
+
+  Signin(){
```

Comment:
Why do you have it this way?

Reply:
if I do not, I get this error:
![screenshot-2018-6-27 hot corner delicacies](https://user-images.githubusercontent.com/28805113/41971934-d5389fa8-7a18-11e8-8119-7f66c4af21dd.png)